extensions/csv/src/main/java/io/deephaven/csv/tokenization/external/DoubleParser.java

```diff
@@ -0,0 +1,612 @@
+package io.deephaven.csv.tokenization.external;
+
+/*
+ * @(#)FastDoubleParser.java
+ * Copyright © 2021. Werner Randelshofer, Switzerland. MIT License.
+ */
+
+/*
+ * Modifications by kosak: change package name, and use booleans to indicate errors
+ * rather than throwing exceptions, change method name to tryParseDouble, and accept
+ * our ByteSlice type rather than a byte[] array.
+ */
+
+import io.deephaven.csv.containers.ByteSlice;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * This is a C++ to Java port of Daniel Lemire's fast_double_parser.
+ * <p>
+ * The code has been changed, so that it parses the same syntax as
+ * {@link Double#parseDouble(String)}.
```

Comment:
At the very least, I'd want to make sure that DoubleParser passes https://github.com/srisatish/openjdk/blob/master/jdk/test/java/lang/Double/ParseDouble.java

Reply:
I wouldn't think it's in scope for us to unit test someone else's library, but we *can* do it...

Reply:
I'm happy to create a fork of https://github.com/wrandelshofer/FastDoubleParser and have unit tests live there (and potentially contribute back) if they don't exist already. As a way to bulk up any unit tests, and have it live with the source library. (It would also be nice to publish it instead of bringing it into the codebase.)

Most of the bugs for https://github.com/wrandelshofer/FastDoubleParser/issues?q=is%3Aissue+is%3Aclosed seem to be around differences from the original fast double parser or differences from JDK parsing... from our side, I'd really want an easy switch back to JDK parsing.